src/gitdb.ts

```diff
@@ -37,127 +38,108 @@ class GitDB {
   static async init(config: Config): Promise<GitDB> {
     const gitDb = new GitDB(config);
 
-    if (gitDb.config.cache) {
-      const collectionDirectoryNames = await fs.readdir(config.dbDir);
-      const collectionPromises = collectionDirectoryNames.map(
-        async (collectionName) => {
-          await gitDb.readCollection(collectionName, true);
-        },
-      );
-      await Promise.all(collectionPromises);
-    }
+    const collectionDirectoryNames = await fs.readdir(config.dbDir);
+    const collectionPromises = collectionDirectoryNames.map(
+      async (collectionName) => {
+        gitDb.createCollection(collectionName);
+      },
+    );
+    await Promise.all(collectionPromises);
     return gitDb;
   }
 
-  async readCollection(
-    collectionName: string,
-    cache: boolean,
-  ): Promise<Collection<any>> {
+  public async readDocuments(collectionName: string): Promise<any[]> {
     const collectionPath = `${this.config.dbDir}/${collectionName}`;
     const documentNames = await fs.readdir(collectionPath);
 
-    const collection = cache
-      ? this.collections[collectionName]
-      : new Collection(this, collectionName);
+    const documentPromises = documentNames.map(this.readFile(collectionPath));
```

Comment:
if we have `utils/file.js` this method might need an update

Reply:
could you explain in more detail?

Reply:
remove gitDb from FIlestrategy
move fileOperations on separate file
use same interface for both strategies (Collection strategy Interface)